src/controllers/useAddingTodo.ts

```diff
@@ -0,0 +1,45 @@
+import { useState, useCallback } from 'react';
+import { createTodo } from '../api/todos';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  showError: ((message: string) => void);
+};
+
+type HookOutput = [
+  boolean,
+  Todo | null,
+  (todoInfo: Omit<Todo, 'id'>) => void,
+];
```

Comment:
👍

Reply:
I guess it's safer to return an object if you have 3 or more things to return. You will not be messed up with the order.

Reply:
Good point! Thanks :)